improver/utilities/temporal.py

```diff
@@ -489,30 +504,52 @@ def check_input_cube_time(self, input_cube: Cube, local_time: datetime) -> None:
         # timezone_cube.coord("UTC_offset") is monotonically increasing. It needs to be
         # decreasing so that the required UTC time for local_time will be increasing
         # when it is calculated.
+        enforce_coordinate_ordering(
+            self.timezone_cube, ["UTC_offset"], anchor_start=False
+        )
         self.timezone_cube = self.timezone_cube[:, :, ::-1]
+
         timezone_coord = self.timezone_cube.coord("UTC_offset")
         timezone_coord.convert_units("seconds")
         output_times = [
             local_time - timedelta(seconds=np.int(offset))
             for offset in timezone_coord.points
         ]
         if input_time_points != output_times:
+            if input_cube.coord("time").bounds is not None:
```

Comment:
How does the `if input_time_points != output_times` work for a partial period case? I would have expected the timezone cube to contain all possible UTC offsets, whilst the `input_time_points` would only contain a partial period, and therefore the `if input_time_points != output_times` check would be true, however, this doesn't seem to be how the `test_partial_period_update` test is set up given that `input_time_points == output_times`. Is this correct?

Reply:
Your question, as discussed, gets to the heart of the limitations of these changes. At present the partial period update still requires that there be at least some valid data in every timezone. This means that eastern timezones can be updated all day down to a single remaining hour. However, western timezones will get far fewer updates through the first day of the forecast; once the east is in a different day we stop updating the west.

It is obviously required that all timezones can be updated throughout the first day to ensure consistency between the hourly symbols and the summary. As such another ticket has been created to go back and make further modifications to this code to enable this behaviour. This may require that masked data is returned for eastern timezones, or that the final hour for each is used to populate the summary once the day has passed in these timezones.

The fact that this behaviour doesn't yet exist is why this if statement is currently as it is.